darwin/exporter/formats/darwin.py

```diff
@@ -16,65 +16,144 @@
-                    }
-                ],
-                "image": {
-                    "filename": "a_file_name.json",
-                    "height": 1000,
-                    "width": 2000,
-                    "url": "https://www.darwin.v7labs.com/..."
-                }
-            }
+        A dictionary with the annotations in Darwin v2 format.
     """
-    annotations: List[Dict[str, Any]] = []
-    print(annotations)
+    annotations_list: List[Dict[str, Any]] = []
+
     for annotation in annotation_file.annotations:
-        payload = {
-            annotation.annotation_class.annotation_type: _build_annotation_data(
-                annotation
-            ),
-            "name": annotation.annotation_class.name,
-        }
+        annotation_data = _build_v2_annotation_data(annotation)
+        annotations_list.append(annotation_data)
+
+    slots_data = _build_slots_data(annotation_file.slots)
+    item = _build_item_data(annotation_file)
+    item["slots"] = slots_data
+
+    return {
+        "version": "2.0",
+        "schema_ref": "https://darwin-public.s3.eu-west-1.amazonaws.com/darwin_json/2.0/schema.json",
+        "item": item,
+        "annotations": annotations_list,
+    }
+
 
-        if (
-            annotation.annotation_class.annotation_type == "complex_polygon"
-            or annotation.annotation_class.annotation_type == "polygon"
-        ) and "bounding_box" in annotation.data:
-            payload["bounding_box"] = annotation.data["bounding_box"]
+def _build_v2_annotation_data(annotation: dt.Annotation) -> Dict[str, Any]:
+    annotation_data = {"id": annotation.id, "name": annotation.annotation_class.name}
+    if annotation.annotation_class.annotation_type == "bounding_box":
+        annotation_data["bounding_box"] = _build_bounding_box_data(annotation.data)
+    elif annotation.annotation_class.annotation_type == "tag":
+        annotation_data["tag"] = {}
+    elif annotation.annotation_class.annotation_type == "polygon":
+        polygon_data = _build_polygon_data(annotation.data)
+        annotation_data["polygon"] = polygon_data
+        annotation_data["bounding_box"] = _build_bounding_box_data(annotation.data)
+    return annotation_data
 
-        annotations.append(payload)
 
+def _build_bounding_box_data(data: Dict[str, Any]) -> Dict[str, Any]:
+    if "bounding_box" in data:
+        data = data["bounding_box"]
     return {
-        "annotations": annotations,
-        "image": {
-            "filename": annotation_file.filename,
-            "height": annotation_file.image_height,
-            "width": annotation_file.image_width,
-            "url": annotation_file.image_url,
+        "h": data.get("h"),
+        "w": data.get("w"),
+        "x": data.get("x"),
+        "y": data.get("y"),
+    }
+
+
+def _build_polygon_data(
+    data: Dict[str, Any]
+) -> Dict[str, List[List[Dict[str, float]]]]:
```

Comment:
This looks kind of nebulous and not immediately understandable. Can we create a datatype here instead
`Path = List[List[Dict[str, float]]]` or something similar.

Reply:
if it didn't break a bunch of other typing I'd even suggest more structure here with namedtuples, like
```python
def Point(NamedTuple):
    x: float
    y: float
    
Path = List[Point]
Paths = List[Path]
```

We're exploring stronger typing for darwin future so something like that'll probably make it in.

Reply:
I am in favor of this, you want to add it into this PR or should it be a ticket for darwin-py after this is merged?

Reply:
For now can you add this to the PR just as 
```python
darwin/datatypes.py
Path = List[Dict[str, float]]
Paths = List[Path]

def _build_polygon_data(...) -> Dict[str, Paths]
```

I'll do the stronger typing stuff with darwin future code, but the changes above will be backwards compatible with existing typing, it just explicitly names them for easier understanding.

Reply:
@Nathanjp91 , how about PolygonPath and PolygonPaths, since "Path" collides with the commonly used from pathlib import Path